### PR TITLE
[Internal] CosmosElement: Add Code Generator for CosmosNumbers

### PR DESCRIPTION
**Pull request by bchong95**
# Code Generator for CosmosNumbers

## Description

Adding Code Generator for CosmosNumbers. This is to provide consistency in format and also for easy additions in the future.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.cs [hunk 1]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.cs]: Should this be deleted?

File: Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.tt [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+Dictionary<string, string> cosmosNumberTypeToDotNetType = new Dictionary<string, string>()
+{
+    { "Number64", "Number64" },
+    { "Int8", "sbyte" },
+    { "Int16", "short" },
+    { "Int32", "int" },
+    { "Int64", "long" },
+    { "UInt32", "uint" },
+    { "Float32", "float" },
+    { "Float64", "double" },
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.tt]: What about ulong and ushort?

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.tt]: they aren't used in the extended type system.

File: Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.cs]: Please explain why this empty file exists.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/CosmosElements/Numbers/CosmosNumberCodeGenerator.cs]: The generator writes to multiple files and this one by default. The pipe closes before the generator writes any bytes to it, so an empty file gets produced.

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 3]
```diff
@@ -103,6 +108,17 @@
...
 
+  <ItemGroup>
+    <None Update="CosmosElements\Numbers\CosmosNumberCodeGenerator.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>CosmosNumberCodeGenerator.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:119]: What does it represent?

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:119]: idk it seems it got auto generated

**Pull request merged**